tests-ng/test_dmesg.py

```diff
@@ -0,0 +1,79 @@
+import re
+from pathlib import Path
+
+import pytest
+
+CONFIG = {
+    "gardener": {
+        "file_path": "/etc/sysctl.d/40-allow-nonroot-dmesg.conf",
+        "restricted": False,
+    },
+    "server and not gardener": {
+        "file_path": "/etc/sysctl.d/40-restric-dmesg.conf",
+        "restricted": True,
+    },
+    "stig": {"file_path": "/etc/sysctl.d/99-stig.conf", "restricted": True},
+}
+
+# server adds /etc/sysctl.d/40-restric-dmesg.conf, gardener excludes it
+#    and adds /etc/sysctl.d/40-allow-nonroot-dmesg.conf instead
```

Comment:
While I get where you are going with this, e.g. not having multiple very similar tests and drying this out, I would vote to align with https://github.com/gardenlinux/gardenlinux/blob/5c964f41922e379f4e0fd06012f55463911b1968/tests-ng/DEVELOPER.md#parametrization and use clear (but multiple tests) that use `@pytest.mark.feature`. This way one does not have to understand the `CONFIG` map and its logic and can _just read_ the test.

Reply:
> The alternative would be 3 screens of code and jumping back and forth to find the most important information.

Not sure if I follow. If I understand it correctly, Eike's suggestion is to turn this one parameterised test into three tests, does not look like each of them would be a screen of code.

I don't have strong feelings in this case, sound like an edge case where I would be willing to accept both solutions, but my gut feeling would be to prefer the three individual tests because reading the parameter and how it extracts information from the dictionary does seem non-trivial to me.

But this would not be something where I insist on a change to approve, I feel.

Does that help?